host_core/lib/host_core/providers/builtin/numbergen.ex

```diff
@@ -1,17 +1,17 @@
 defmodule HostCore.Providers.Builtin.Numbergen do
   def invoke("GenerateGuid", _payload) do
-    Msgpax.pack!(UUID.uuid4())
+    IO.iodata_to_binary(Msgpax.pack!(UUID.uuid4()))
   end
 
   def invoke("RandomInRange", payload) do
     params = Msgpax.unpack!(payload)
 
-    min = min(params["min"], 0)
-    max = max(params["max"], 4_294_967_295)
-    Msgpax.pack!(Enum.random(min..max))
+    min = max(params["min"], 0)
+    max = min(params["max"] - 1, 4_294_967_295)
+    IO.iodata_to_binary(Msgpax.pack!(Enum.random(min..max)))
```

Comment:
So our random generator is in the range of `[0..MAX)`, and can't be the max number? The only thing that seems weird to me here is that if I supply `4_294_967_295` as the max number, then the highest number it will generate is one less than that (`4_294_967_294`), but if I supply a bigger number like `4_999_999_999` it will generate up to the max U32

I'd suggest removing the subtracting 1, since `Enum.random/1` is inclusive on both ends